src/source/to-shift.js

```diff
@@ -1,31 +1,98 @@
 'use strict';
 
-function removeFunctionContents(obj) {
-    obj.type = obj.type.replace(/^(?:Eager|Lazy)/, '');
+function tryRemoveFunctionContents(obj) {
+    let m = obj.type.match(
+        /^(?:Eager|Lazy)(Function(?:Expression|Declaration)|Method|[GS]etter|(ArrowExpression)With(?:FunctionBody|Expression))$/
```

Comment:
I'd be a little more comfortable with a big `switch`. While it's probably a bit slower, it's more readable and more searchable.

Reply:
Initially I used regex just for trimming the prefix part (it was `/^(?:Eager|Lazy)/`), but later decided that it's easy enough to put types here as well. Can revert.

Reply:
Ah, I recalled why I changed the switch - I didn't really like duplication of each node x2 for both `Eager` and `Lazy` variants, feels hard to read too.

What if I just expand regex to list all types fully instead of using subexpressions?

Or, I guess, I could strip `Eager` / `Lazy` first and then do the switch on the suffix part?

Reply:
> Or, I guess, I could strip Eager / Lazy first and then do the switch on the suffix part?

I went with this approach, hopefully it's a good enough balance for readability.

Reply:
*sigh* this broke something, need to recheck.

Reply:
Fixed it.